src/utils/joi.ts

```diff
@@ -1,8 +1,12 @@
 import { Request } from 'express';
 import * as Joi from 'joi';
+import mongoose from 'mongoose';
 import wrapMiddleware from './express';
 
-export const JoiMongoObjectId = Joi.string().hex().length(24);
+export const JoiObjectId = Joi.string()
+    .hex()
+    .length(24)
+    .custom((value) => new mongoose.Types.ObjectId(value));
```

Comment:
We should work with strings in gateway

```suggestion
export const JoiMongoObjectId = Joi.string().hex().length(24);
```

Reply:
done